blockbot/blockbot.php

```diff
@@ -80,6 +80,13 @@ function blockbot_init_1(App $a) {
 	if (!Config::get('blockbot', 'good_crawlers')) {
 		$agents = array_merge($agents, $good_agents);
 	}
+	else{
```

Comment:
Code standards: Please refer to the PSR-12 coding standards for formatting:
- `else` should be on the same line as the closing brace
- a space after `foreach`
- a space between condition and opening brace

See https://www.php-fig.org/psr/psr-12/

Reply:
Oh sorry, I'm not very good at php coding style. I'm trying to find linter in VSCode. I find one using Code_Sniffer (phpcs) and set it to PSR12. Unfortunately it does more than linting it applies change directly. When it reformats it also applies 4 spaces instead of Tab. It looks like for Friendica you are using tab ?

Reply:
Yes, this is the only exception to the standard.

Reply:
You can just fix the 3 issues I mentioned above and be done with it for this PR.

Reply:
I've just sent it ;-)

Looks like if I want to contribute more often, I need to find a good way to tune my IDE. Here I use VSCode but maybe I have to find another one because there is no extension which does only linting. It is always fixer which applies spaces.

So I test phpcs and php-cs-fixer based extension but doesn't offer the good solution.

Reply:
VSCode is working with Friendica. I'm partially using it as well.

Reply:
@MrPetovan For phpcs you can supply a config file with the exceptions to the standards. I believe you can make it ignore the tab vs space issue project-wise. Watch out though, not all the files are compliant, so you might fix more than your bargained for when you edit many Friendica files outside of `src/`.

Reply:
Does VSCode not support the `editorconfig` rule set? It would not help with the line break, but the `TAB` should be set correctly by it.

Reply:
We're talking about two different things. VSCode probably supports both tab and space indentation, but @valvin1 is talking about a third-party PHP linter that follows PSR-12 to the letter by default, which includes switching the indentation to spaces for the related code standards fixer. But you can add a config file to exclude specific code standard rules from the detection/fix.

Reply:
Yes, editorconfig is working fine. It's more about linter. Finally I found a solution using phpcs extension and ruleset.xml from friendica repo. But I'm still having many linter error / warning ;-)

Reply:
It's an ongoing effort 😅